core/src/main/java/com/datastax/oss/driver/internal/core/session/DefaultSession.java

```diff
@@ -107,11 +106,50 @@ private DefaultSession(
     this.metricUpdater = context.metricUpdaterFactory().newSessionUpdater();
   }
 
-  private CompletionStage<CqlSession> init(CqlIdentifier keyspace) {
+  public CompletionStage<CqlSession> init(CqlIdentifier keyspace) {
+    DriverConfigProfile defaultConfig = context.config().getDefaultProfile();
+    if (defaultConfig.isDefined(DefaultDriverOption.STARTUP_CUSTOM_BANNER)) {
+      printCustomBanner(defaultConfig.getString(DefaultDriverOption.STARTUP_CUSTOM_BANNER));
+    }
+    if (defaultConfig.getBoolean(DefaultDriverOption.STARTUP_PRINT_BASIC_INFO)) {
+      printBasicStartupInfo();
+    }
     RunOrSchedule.on(adminExecutor, () -> singleThreaded.init(keyspace));
     return singleThreaded.initFuture;
   }
 
+  protected void printCustomBanner(String banner) {
+    STARTUP_LOG.info(banner);
+  }
+
+  protected void printBasicStartupInfo() {
+    try {
+      DriverInfo driverInfo = getDriverInfo();
+      int processId = -1;
+      if (Native.isGetProcessIdAvailable()) {
+        processId = Native.getProcessId();
+      }
+      long start = System.nanoTime();
+      STARTUP_LOG.info("[{}] {} initializing (PID: {})", logPrefix, driverInfo, processId);
+      singleThreaded.initFuture.thenRun(
+          () -> {
+            Duration elapsed =
+                Duration.ofMillis(TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start));
+            String elapsedStr =
```

Comment:
Just a pretty-print for durations. It is a common enough problem that I think we should have an utility method somewhere to print durations in good ol' English. Currently our log messages print duration in ISO-8601 format, which is not very user-friendly.

Reply:
I wrote `NanoTime.format` to that effect for token map logs.